mesonbuild/build.py

```diff
@@ -1952,6 +1957,8 @@ def post_init(self) -> None:
                     self.suffix = 'rlib'
                 elif self.rust_crate_type == 'staticlib':
                     self.suffix = 'a'
+                elif self.rust_crate_type == 'object':
+                    self.suffix = 'o'
```

Comment:
On windows, with msvc compatibly linkers this needs to be `.obj` I think.

Reply:
I believe rustc uses `.o` even for MSVC.

Reply:
Indeed: https://github.com/rust-lang/rust/issues/37207

Reply:
We should probably have a comment here then, or someone else (or me after forgetting) will change this, meaning well :)

Reply:
added